### PR TITLE
rec/auth: Refactor version reporting code and write version to stdout

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Refactor is needed to implement #14455.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by Habbie — APPROVED** [review]
Looks good. I do wonder if instead of parsing lines from a string we just generated from a bunch of lines, we could instead pass the vector around more, but I think we might just be moving the pain around then.

**Comment by omoerbeek** [comment]
> Looks good. I do wonder if instead of parsing lines from a string we just generated from a bunch of lines, we could instead pass the vector around more, but I think we might just be moving the pain around then.

I thought about this as well, and came to the same conclusion. Plus: this is code that is executed once, so no big deal from a performance point of view.

**Comment by omoerbeek** [comment]
Hmm, I might also have hit a case where various versions of clang-format do not agree.

File: pdns/version.hh [hunk 1]
```diff
@@ -22,14 +22,19 @@
...
-string compilerVersion();
-void showProductVersion();
-void showBuildConfiguration();
-string fullVersionString();
-string getPDNSVersion();
-string productName();
-string productTypeApiType();
-void versionSetProduct(ProductType pt);
-ProductType versionGetProduct();
+[[nodiscard]] std::string compilerVersion();
```
Comment (Habbie) [inline pdns/version.hh:31]: I guess these don't need `std::` now, but I don't mind

**Review by Habbie — APPROVED** [review]

**Pull request merged**